From cfda9e196b7087ce69f603226d94d1161a9af8c1 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Tue, 25 Mar 2025 08:29:55 +0000
Subject: [PATCH] Fix build with gcc-15

:Release Notes:
Add missing include.

:Detailed Notes:
Fixes:

http://gecko.lge.com:8000/Errors/Details/1144153
TOPDIR/BUILD/work/raspberrypi4_64-webos-linux/armnn/22.08/git/profiling/common/include/Packet.hpp:29:20: error: expected ')' before 'header'
   29 |     Packet(uint32_t header)
      |           ~        ^~~~~~~
      |                    )
...

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRR-5443] Create GPVB with Yocto 5.2 Walnascar
---
Upstream-Status: Pending

 profiling/client/include/ISendTimelinePacket.hpp | 1 +
 profiling/common/include/Packet.hpp              | 1 +
 third-party/cxxopts/cxxopts.hpp                  | 1 +
 3 files changed, 3 insertions(+)

diff --git a/profiling/client/include/ISendTimelinePacket.hpp b/profiling/client/include/ISendTimelinePacket.hpp
index 4785e04ef..3ee5b6b96 100644
--- a/profiling/client/include/ISendTimelinePacket.hpp
+++ b/profiling/client/include/ISendTimelinePacket.hpp
@@ -6,6 +6,7 @@
 #pragma once
 
 #include <algorithm>
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/profiling/common/include/Packet.hpp b/profiling/common/include/Packet.hpp
index d8fa2709e..93a3b7b6c 100644
--- a/profiling/common/include/Packet.hpp
+++ b/profiling/common/include/Packet.hpp
@@ -7,6 +7,7 @@
 
 #include "ProfilingException.hpp"
 
+#include <cstdint>
 #include <memory>
 
 namespace arm
diff --git a/third-party/cxxopts/cxxopts.hpp b/third-party/cxxopts/cxxopts.hpp
index 5cebc4bd1..78831e156 100644
--- a/third-party/cxxopts/cxxopts.hpp
+++ b/third-party/cxxopts/cxxopts.hpp
@@ -28,6 +28,7 @@ THE SOFTWARE.
 #define CXXOPTS_HPP_INCLUDED
 
 #include <cctype>
+#include <cstdint>
 #include <cstring>
 #include <exception>
 #include <iostream>
